### ISSUE TITLE
How many hours to pretrain open-dcoder-0.5B

### ISSUE THREAD

**Issue by AIxyz** [pengzhangzhi/Open-dLLM#21]
I want to pretrain a model like https://huggingface.co/fredzzp/open-dcoder-0.5B by https://huggingface.co/datasets/fredzzp/fine_code
I use https://github.com/pengzhangzhi/Open-dLLM/blob/main/configs/pretrain/qwen2_5_coder_500M.yaml

I use 4*H800-80G, how many hours to pretrain?

By the way, what is the train_size of https://huggingface.co/fredzzp/open-dcoder-0.5B?

From the loss step(total 190k steps) in https://oval-shell-31c.notion.site/Open-dLLM-Open-Diffusion-Large-Language-Model-25e03bf6136480b7a4ebe3d53be9f68a , I guess the train_size is 400_000_000_000, because I use 10.5h to finish 477 steps with train_size 1_000_000_000 in 4*H800-80G

```bash
ln -s /mnt/tenant-home_speed/shared/models/huggingface/Qwen2.5-Coder-0.5B--e793650-C7 /workdir/base_model
ln -s /mnt/tenant-home_speed/shared/data/huggingface/fine_code--LFS/data /workdir/data

torchrun --nnodes=$NNODES --nproc-per-node $NPROC_PER_NODE --node-rank $NODE_RANK \
  --master-addr=$MASTER_ADDR --master-port=$MASTER_PORT tasks/train_torch.py \
  configs/pretrain/qwen2_5_coder_500M.yaml \
  --model.model_path=/workdir/base_model \
  --data.train_path=/workdir/data \
  --data.max_seq_len=4096 \
  --data.train_size=10_000_000_000 \
  --train.ckpt_manager=dcp \
  --train.micro_batch_size=8 \
  --train.global_batch_size=512 \
  --train.output_dir=/workdir/outputs/Coder-0.5B_mdm \
```

**Comment by pengzhangzhi** [comment]
ya correct, 4h800 would need like more than a month.
